src/context/MetricsContext/MetricsContext.tsx

```diff
@@ -0,0 +1,137 @@
+import { isDevelopment } from "config/env";
+import { useSubaccountAddress } from "context/SubaccountContext/SubaccountContext";
+import { useOracleKeeperFetcher } from "domain/synthetics/tokens";
+import { useChainId } from "lib/chains";
+import { getAppVersion } from "lib/version";
+import { getWalletNames } from "lib/wallets/getWalletNames";
+import { Context, PropsWithChildren, useMemo, useRef } from "react";
+import { createContext, useContextSelector } from "use-context-selector";
+import mapValues from "lodash/mapValues";
+
+export type MetricsContextType = {
+  sendMetric: (params: {
+    event: string;
+    fields?: Record<string, any>;
+    time?: number;
+    isError: boolean;
+    message?: string;
+  }) => void;
+  setPendingEvent: <T>(metricId: string, eventData: T) => T;
+  getPendingEvent: (metricId: string, clear?: boolean) => any;
+  startTimer: (metricId: string) => void;
+  getTime: (metricId: string, clear?: boolean) => number | undefined;
+};
+
+const context = createContext<MetricsContextType | null>(null);
+
+export function MetricsContextProvider({ children }: PropsWithChildren) {
+  const { chainId } = useChainId();
+  const fetcher = useOracleKeeperFetcher(chainId);
+  const subaccountAddress = useSubaccountAddress();
+
+  const pendingEvents = useRef({});
+  const timers = useRef({});
+
+  const value: MetricsContextType = useMemo(() => {
+    const setPendingEvent = (metricId: string, eventData: any) => {
+      pendingEvents.current[metricId] = { metricId, ...eventData };
+
+      return eventData;
+    };
+
+    const getPendingEvent = (metricId: string, clear?: boolean) => {
```

Comment:
getMetricData/setMetricData? not sure if Cache also should be mentioned in name anyhow